torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -427,6 +436,126 @@ TensorView* TensorView::computeAt(
   return this;
 }
 
+void TensorView::computeWith(
+    TensorView* consumer,
+    int consumer_pos,
+    bool best_effort) {
+  TORCH_INTERNAL_ASSERT(
+      !container()->isA<kir::Kernel>(),
+      "Function invalid for kernel container.");
+
+  if (isFusionInput()) {
+    return;
+  }
+
+  auto consumers = ir_utils::consumerTvsOf(this);
+  TORCH_CHECK(
+      std::find(consumers.begin(), consumers.end(), consumer) !=
+          consumers.end(),
+      "Not a direct consumer: ",
+      consumer->toString());
+
+  if (consumer_pos < 0) {
+    consumer_pos += int(consumer->nDims()) + 1;
+  }
+
+  TORCH_INTERNAL_ASSERT(
+      consumer_pos >= 0 && consumer_pos <= nDims(),
+      "Invalid inline position for ",
+      consumer->toString(),
+      ": ",
+      consumer_pos);
+
+  // Find the corresponding position in this tensor
+  int this_pos = 0;
+  if (consumer_pos > 0) {
+    auto c2p = BestEffortReplay::replayPasC(
+                   this, consumer, -1, PairwiseRootDomainMap(this, consumer))
+                   .getReplay();
+    // Find the innermost matching producer ID
+    for (int consumer_offset = consumer_pos - 1; consumer_offset >= 0;
+         --consumer_offset) {
+      auto consumer_id = consumer->axis(consumer_offset);
+      auto c2p_it = c2p.find(consumer_id);
+      if (c2p_it == c2p.end()) {
+        continue;
+      }
+      auto producer_id = c2p_it->second;
+      this_pos = std::distance(
+                     domain()->domain().begin(),
+                     std::find(
+                         domain()->domain().begin(),
+                         domain()->domain().end(),
+                         producer_id)) +
+          1;
+      break;
+    }
+  }
+
+  const auto max_inline_pos =
+      MaxPosCalculator({}, true).getMaxPosAll(this, best_effort);
+
+  if (best_effort) {
+    this_pos = std::min<int>(max_inline_pos, this_pos);
+  }
+
+  // hoist inner most broadcast
+  while (this_pos > 0 && axis(this_pos - 1)->isBroadcast()) {
+    this_pos--;
+  }
+
+  TORCH_CHECK(
+      this_pos <= max_inline_pos,
+      "Invalid computeWith position for T",
+      name(),
+      ": ",
+      this_pos,
+      ". Maximum allowed value:",
+      max_inline_pos);
+
+  // The position must be right of the computeAt position
+  TORCH_CHECK(
+      this_pos >= getComputeAtPosition(),
+      "Position must be right of the computeAt position. Position: ",
+      this_pos,
+      ", computeAt position: ",
+      getComputeAtPosition());
```

Comment:
I think this should be
```C++
if (this_pos < getComputeAtPosition()) {
  return;
}
```
so that it is consistent to the behavior below?
```
  // If it's already set to be computed with the consumer and the
  // position is higher, nothing to change
```

Reply:
I think we shouldn't simply ignore a position that is left of the computeAt position, but that's just my feeling. I'll update the comment.

Reply:
Comment updated